### PR TITLE
Add one_done trigger rule

### LINKED ISSUE #26099: Add one_done trigger rule
### Body

Action: trigger as soon as 1 upstream task is in success or failuire
This has been requested in https://stackoverflow.com/questions/73501232/how-to-implement-the-one-done-trigger-rule-for-airflow
I think this can be useful for the community.

**The Task:**
Add support for new trigger rule `one_done`
You can use as reference previous PRs that added other trigger rules for example: https://github.com/apache/airflow/pull/21662

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by pankajastro**
closes: https://github.com/apache/airflow/issues/26099

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by raphaelauv** [comment]
`one_done` -> `at_least_one_done`

cause `one_done` mean one and only one upstream task must be done 

wdyt ?

**Comment by uranusjr** [comment]
We already have `one_success` and `one_failed`, IMO `one_done` fits better.

File: tests/ti_deps/deps/test_trigger_rule_dep.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def test_one_failure_tr_success(self, get_task_instance):
         dep_statuses = tuple(
             TriggerRuleDep()._evaluate_trigger_rule(
                 ti=ti,
-                successes=0,
+                successes=1,
```
Comment (eladkal) [inline tests/ti_deps/deps/test_trigger_rule_dep.py]: Why do we need to change test of another trigger rule?

File: tests/ti_deps/deps/test_trigger_rule_dep.py [hunk 2]
```diff
@@ -163,6 +163,26 @@ def test_one_failure_tr_success(self, get_task_instance):
         )
         assert len(dep_statuses) == 0
 
+    def test_one_done_tr_success(self, get_task_instance):
```
Comment (eladkal) [inline tests/ti_deps/deps/test_trigger_rule_dep.py:166]: Can we have more tests to cover the full functionality of the trigger rule?
We have here 1 test for success..
We need also 1 for skip, 1 for upstream failed I think?

Comment (pankajastro) [reply tests/ti_deps/deps/test_trigger_rule_dep.py:166]: Added more tests

File: tests/ti_deps/deps/test_trigger_rule_dep.py (hunk 1 above)
Comment (pankajastro) [reply tests/ti_deps/deps/test_trigger_rule_dep.py]: I think by mistake I made this change. I have reverted it. thanks for pointing it out.

**Comment by potiuk** [comment]
Can someone more involved in trigger rules take a look at this one? @uranusjr @ashb  - seems good to me in general, but maybe it can be done simpler :)

**Review by kaxil — APPROVED** [review]

**Pull request merged**